multiversx_sdk/network_providers/http_resources.py

```diff
@@ -30,3 +37,132 @@ def vm_query_response_to_smart_contract_query_response(raw_response: Dict[str, A
+    result.round = response.get("round", 0)
+    result.epoch = response.get("epoch", 0)
+    result.value = response.get("value", 0)
+
+    sender = response.get("sender", "")
+    result.sender = Address.new_from_bech32(sender) if sender else EmptyAddress()
+
+    receiver = response.get("receiver", "")
+    result.receiver = Address.new_from_bech32(receiver) if receiver else EmptyAddress()
+
+    result.gas_price = response.get("gasPrice", 0)
+    result.gas_limit = response.get("gasLimit", 0)
+
+    data = response.get("data", "") or ""
+    result.function = response.get("function", "")
+
+    result.data = base64.b64decode(data).decode()
+    result.status = TransactionStatus(response.get("status"))
+    result.timestamp = response.get("timestamp", 0)
+
+    result.block_nonce = response.get("blockNonce", 0)
+    result.hyperblock_nonce = response.get("hyperblockNonce", 0)
+    result.hyperblock_hash = response.get("hyperblockHash", "")
+
+    result.logs = transaction_logs_from_request(response.get("logs", {}))
+    result.raw_response = response
+
+    return result
+
+
+def transaction_logs_from_request(raw_response: dict[str, Any]) -> TransactionLogs:
+    address = raw_response.get('address', "")
+
+    events = raw_response.get('events', [])
+    events = [transaction_events_from_request(event) for event in events]
+
+    return TransactionLogs(
+        address=address,
+        events=events
+    )
+
+
+def transaction_events_from_request(raw_response: dict[str, Any]) -> TransactionEvent:
+    address = raw_response.get('address', '')
+    identifier = raw_response.get('identifier', '')
+    topics = [base64.b64decode(topic.encode()) for topic in raw_response.get('topics', [])]
+
+    raw_data = base64.b64decode(raw_response.get('responseData', "").encode())
+
+    data = raw_response.get("data", None)
+    if data is not None:
+        data = base64.b64decode(data.encode())
+    else:
+        data = b""
+
+    additional_data = raw_response.get("additionalData", None)
+    additional_data = [base64.b64decode(data.encode()) for data in additional_data] if additional_data is not None else []
+
+    if len(additional_data) == 0:
+        if raw_data:
+            additional_data.append(raw_data)
+
+    return TransactionEvent(
+        raw=raw_response,
+        address=address,
+        identifier=identifier,
+        topics=topics,
+        data=data,
+        additional_data=additional_data
+    )
+
+
+def smart_contract_result_from_api_request(raw_response: dict[str, Any]) -> SmartContractResult:
+    sc_result = _smart_contract_result_from_request(raw_response)
+    data = raw_response.get("data", "")
+    sc_result.data = base64.b64decode(data.encode())
+    return sc_result
+
+
+def smart_contract_result_from_proxy_request(raw_response: dict[str, Any]) -> SmartContractResult:
+    sc_result = _smart_contract_result_from_request(raw_response)
+    sc_result.data = raw_response.get("data", "").encode()
+    return sc_result
+
```

Comment:
It would have been so nice to have the same response from Proxy and API, but that would be a complicated breaking change :see_no_evil: 

Or maybe the API can implement something like `/transactions/{hash}?pleaseReturnDataOfContractResultsAsDecodedBase64=true`.